mso/resource_mso_site.go

```diff
@@ -448,10 +501,9 @@ func resourceMSOSiteRead(d *schema.ResourceData, m interface{}) error {
 		if con.Exists("maintenanceMode") {
 			d.Set("maintenance_mode", con.S("maintenanceMode").Data().(bool))
 		}
-		if _, ok := d.GetOk("login_domain"); ok {
-			regex := regexp.MustCompile(`apic#(.*)\\{4}(.*)`)
-			unameStr := models.StripQuotes(con.S("username").String())
-			matches := regex.FindStringSubmatch(unameStr)
+		if _, ok := d.GetOk("username"); ok {
```

Comment:
move just after line 488 and the condition should be :

if username {


or something similar.

Reply:
if username != ""